test/test_client.py

```diff
@@ -42,6 +43,28 @@ def test_urlencode(self):
         self.assertEqual("address=%3DSydney+~", encoded_params)
 
     @responses.activate
+    def test_queries_per_second(self):
+        # This test assumes that the time to run a mocked query is
+        # relatively small, eg a few milliseconds. We define a rate of
+        # 3 queries per second, and run double that, which should take at
+        # least 1 second but no more than 2.
+        queries_per_second = 3
+        query_range = range(queries_per_second * 2)
+        for _ in query_range:
+            responses.add(responses.GET,
+                          "https://maps.googleapis.com/maps/api/geocode/json",
+                          body='{"status":"OK","results":[]}',
+                          status=200,
+                          content_type="application/json")
+        client = googlemaps.Client(key="AIzaasdf",
+                                   queries_per_second=queries_per_second)
+        start = time.time()
+        for _ in query_range:
+            client.geocode("Sesame St.")
+        end = time.time()
+        self.assertTrue(start + 1 < end < start + 2)
```

Comment:
True... if anything the guarantee is that we execute requests _no faster_ than the specified rate limit, with no guarantee on the upper bound.

Disregard my comment :)